src/plugins/pluginTypes.ts

```diff
@@ -35,6 +35,11 @@ export type IPlugin = {
   // simple regex replacements on the markdown output
   regexMarkdownModifications?: IRegexMarkdownModification[];
 
+  // operations on pages to define the markdown's frontmatter
+  frontmatterTransform?: {

```

Comment:
I'm not completely sold on the `frontmatterTransform` name. Can you think of a better one?

Reply:
How about `frontMatterGenerator`? I'm camel casing "Matter" because `front matter` is normally written as two words.